src/libraries/System.Formats.Tar/src/System/Formats/Tar/TarHeader.Write.cs

```diff
@@ -374,12 +374,10 @@ private int WritePosixName(Span<byte> buffer)
 
             if (_name.Length > FieldLengths.Name)
             {
-                int prefixBytesLength = Math.Min(_name.Length - FieldLengths.Name, FieldLengths.Name);
-                Span<byte> remaining = prefixBytesLength <= 256 ?
-                    stackalloc byte[prefixBytesLength] :
-                    new byte[prefixBytesLength];
-
-                int encoded = Encoding.ASCII.GetBytes(_name.AsSpan(FieldLengths.Name), remaining);
+                int prefixBytesLength = Math.Min(_name.Length - FieldLengths.Name, FieldLengths.Prefix);
```

Comment:
First portion of the bug:
- The name field contains the first 100 bytes of a path.
- The prefix field contains the subsequent 155 bytes of a path. Anything beyond that gets truncated.
- PAX and GNU store the full path, regardless of size, in a metadata entry (extended attributes for PAX, LongPath for GNU).

The bug here was that the prefix byte length was being calculated incorrectly.